app/controllers/identity_controller.rb

```diff
@@ -1,15 +1,76 @@
 # frozen_string_literal: true
 class IdentityController < ApplicationController
+  skip_forgery_protection
+
   def create
     unless FeatureFlag.active?(:identity_open)
       raise IdentityEndpointOffError,
             "Could not access the get an identity endpoint because the " \
               ":identity_open feature flag is off"
     end
 
+    unless valid_identity_params?
+      raise IdentityDataError,
+            "Could not continue because the " \
+              "data from Identity was unable to be verified"
+    end
+
     head :no_content
   end
 
+  private
+
+  def allowed_create_params
+    params.permit(:email, :redirect_uri, :client_title, :journey_id, :sig)
+  end
+
+  # Compare the signature provided by Identity with the signature calcuated
+  # using the recieved params. If they match then data has not been tampered with.
+  def valid_identity_params?
+    return false if expected_signature.blank? || new_signature.blank?
+
+    expected_signature.upcase == new_signature.upcase
+  end
+
+  # Retrieve the list of params required for the paramater check. The keys
+  # must be in alphabetical order.
+  def sorted_identity_params
+    params_without_signature = allowed_create_params.except("sig")
```

Comment:
See the note at https://github.com/DFE-Digital/get-an-identity/blob/main/docs/find-lost-trn-integration.md#computing-the-signature
Ideally this needs to look at all params (not just the ones that are currently known about) for future compatibility.

Reply:
Thanks @gunndabad I did discuss this with @tvararu and if the params are unlikely to be updated very often we would like to use permitted params to have more control of what can data passed in. Is this something that is likely to change much in the future?

Reply:
It will almost certainly change at some point. We can keep as-is for now and review in future.